Aos 2 surface sweep

efifogel opened the pull request:
## Summary of Changes

1. The changes introduced by this branch do not affect the exposed API whatsoever. 
2. I've change the name of the main directory that corresponds to the "2D Intersection of Curves" package from Sweep_line_2 to Surface_sweep_2. It's a better name in general, and it also matches the name of the directory that holds the 2D arrangement code, namely Arrangement_on_surface_2.
3. Sometime ago Andreas and perhaps others moved files from the Aos_2 directory to the Sl_2 (Now Ss_2). The move was done a bit hastily. I've moved one file that was left out, and I moved back some files that shouldn't have been moved in the first place (naturally, updating the included license file.)
4. In the new Ss_2 I've introduced a namespace, namely Surface_sweep_2. However, I'm using it at this point only internally, so the exposed free functions are not affected.
5. The changes where made in order to get rid of the use of 'void *' and castings (both C and C++ styles) which where used to handle inter dependencies. For example, the surface sweep uses types called Event and Subcurve, and each type contains references to object of the other. Now the code has better type checking and I manage to optimize it a bit, so it is more efficient space and time wise.
6. I've also fixed the scheme of instantiation so that the surface-sweep stuff can be exposed (once the appropriate documentation is provided as part of a formal feature submission). For example, the main sweep template is parametrized only by a visitor, and the visitor is parametrized by the traits, the event, the subcurve, and the allocator. (In the past, several types where parametrized by several other and all had defaults, which obviously didn't make sense). In particular, I've fixed the definition of the Allocator so that a user of the sweep can provide her own.
7. I had to use curiously recurring template pattern (CRTP) in several places and they are intermixed. Unfortunately, I cannot use C++11 features, but I managed. Everything compiles on G++ 4.8.4 and VC 14. I have a feeling that I'll have to work on it to get older VC versions to work, we'll see.

## Release Management

* Affected package(s): "2D Intersection of Curves" and "2D Arrangements"
* Issue(s) solved (if any): see above
* Feature/Small Feature (if any): no

sloriot commented:
@efifogel Travis failed with several errors:
- _The following files are referenced in the documentation but do not exist:_ `Surface_sweep_2/doc/Surface_sweep_2/CGAL/Sweep_line_2_algorithms.h`
- Compilation errors in [BSO_2](https://travis-ci.org/CGAL/cgal/jobs/250354791)
There are others packages with compilation errors but I guess they are all related to the latter.

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arr_batched_point_location.h:
```diff
@@ -24,61 +21,75 @@
-  unsigned int                         i = 0;
+  std::vector<Bpl_x_monotone_curve_2> xcurves_vec(arr.number_of_edges());
+  Edge_const_iterator eit;
+  size_t i(0);
```
std::size_t

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arr_bounded_planar_topology_traits_2.h:
```diff
@@ -190,189 +191,162 @@ class Arr_bounded_planar_topology_traits_2 :
+  // the non-C++11 code requires the (re)definition of all constructors of the
+  // derived class. The non-C++11 code follows the commented out C++11 code.
+  // When we move to C++11 we can use the more elgant code.
+#if 0
```
There are some macros in CGAL to enable the code on platforms that support template aliases.

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arr_spherical_topology_traits_2.h:
```diff
@@ -314,157 +306,139 @@ class Arr_spherical_topology_traits_2 {
+  // the non-C++11 code requires the (re)definition of all constructors of the
+  // derived class. The non-C++11 code follows the commented out C++11 code.
+  // When we move to C++11 we can use the more elgant code.
+#if 0
```
Same here

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arr_unb_planar_topology_traits_2.h:
```diff
@@ -165,241 +164,206 @@ class Arr_unb_planar_topology_traits_2 :
+  // the non-C++11 code requires the (re)definition of all constructors of the
+  // derived class. The non-C++11 code follows the commented out C++11 code.
+  // When we move to C++11 we can use the more elgant code.
+#if 0
```
same here

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arrangement_2/Arrangement_on_surface_2_global.h:
```diff
@@ -12,48 +12,51 @@
+#include <boost/type_traits.hpp>
+#include <list>
 
+#include <CGAL/license/Arrangement_on_surface_2.h>
```
The license include directive should be right after the include guards

sloriot commented on Arrangement_on_surface_2/include/CGAL/Arrangement_2/Arrangement_on_surface_2_global.h:
```diff
@@ -182,7 +187,7 @@ void insert(Arrangement_on_surface_2<GeomTraits, TopTraits>& arr,
   }
 */
-  arr_zone.init (c, pl);
+  arr_zone.init(c, pl);
```
Since you are cleaning the files, any reason to keep the part with the _Needs to be deleted!!!_ flag?

sloriot commented on Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_construction_subcurve.h:
```diff
@@ -0,0 +1,220 @@
+  typedef Event*                                        Event_ptr;
+  typedef std::list<unsigned int>                       Halfedge_indices_list;
+
+  /*! Construct deafult. */
```
typo

sloriot commented on Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_insertion_sl_visitor.h:
```diff
@@ -0,0 +1,170 @@
+};
+
+//-----------------------------------------------------------------------------
+// Memeber-function definitions:
```
typo

sloriot commented on Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_no_intersection_insertion_sl_visitor.h:
```diff
@@ -12,66 +12,81 @@
-template <typename Helper_> 
-class Arr_basic_insertion_sl_visitor : public Helper_::Parent_visitor {
+template <typename Helper_, typename Visitor_ = Default>
+class Arr_no_intersection_insertion_sl_visitor :
```
General remark: you might want to replace `_sl_` by `_ss_` everywhere.

sloriot commented on Documentation/doc/Documentation/dependencies:
```diff
@@ -57,7 +57,7 @@ Principal_component_analysis
 Skin_surface_3
 Snap_rounding_2
-Sweep_line_2
+Surface_sweep_2
```
This change will require a redirect in the apache file of the doc so that latest will be correctly redirected.

sloriot commented on Surface_sweep_2/examples/Surface_sweep_2/plane_sweep.cpp:
```diff
@@ -1,18 +1,16 @@
-typedef Traits_2::Curve_2                               Segment_2;
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Arr_segment_traits_2.h>
+#include <CGAL/Surface_sweep_2_algorithms.h>
```
The header renaming is a breaking change that must be announced in Installation.html + the former header should still be kept with a redirect to the new header. You can use macros for this purpose such as in `AABB_tree/include/CGAL/AABB_polyhedron_triangle_primitive.h` for example.

sloriot commented on Surface_sweep_2/include/CGAL/Surface_sweep_2/Curve_pair.h:
```diff
@@ -12,103 +12,84 @@
-      m_pair = std::make_pair (sc2,sc1);
+    // The smallest pointer will be the first.
+    if (sc1 < sc2) m_pair = std::make_pair (sc1,sc2);
+    else m_pair = std::make_pair (sc2,sc1);
```
there is a function `CGAL::make_sorted_pair()` in `CGAL/utility.h`

efifogel commented:
Thanks for the detailed review!
I've fixed all the above except for 2 items:
1. I did not enable the code that uses template aliases, and
2. I did not do anything regarding the need to redirect in the Apache file
of the doc so that latest will be correctly redirected.

In particular, I've introduced

Surface_sweep_2/include/CGAL/Sweep_line_2_algorithms.h

in order to maintain backward compatibility.

Regarding the 2 items I did not do.
1. I couldn't find a macro that conditionally enables template aliases.
I'll be happy to use if exists (or even introduce if not).
2. I'm not familiar with the Apache settings. Is there anything that needs
to be done in the code base?
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 6 July 2017 at 12:03, Sebastien Loriot <notifications@github.com> wrote:

> *@sloriot* commented on this pull request.
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arr_batched_point_location.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125838244>:
>
> >
>    // Go over all arrangement edges and collect their associated x-monotone
>    // curves. To each curve we attach a halfedge handle going from right to
>    // left.
> -  std::vector<Bpl_x_monotone_curve_2>  xcurves_vec(arr.number_of_edges());
> -  Edge_const_iterator                  eit;
> -  unsigned int                         i = 0;
> +  std::vector<Bpl_x_monotone_curve_2> xcurves_vec(arr.number_of_edges());
> +  Edge_const_iterator eit;
> +  size_t i(0);
>
> std::size_t
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arr_bounded_planar_
> topology_traits_2.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125838731>:
>
> > -  typedef Arr_bounded_planar_insertion_helper<InsTraits, Arr, IEvent,
> -                                              ISubcurve>       IHelper;
> -
> -  // Type definition for the batched point-location sweep-line visitor.
> -  typedef Arr_batched_point_location_traits_2<Arr>             BplTraits;
> -  typedef Arr_bounded_planar_batched_pl_helper<BplTraits, Arr> BplHelper;
> +  typedef Arrangement_on_surface_2<Gt2, Self>                   Arr;
> +
> +public:
> +  // The following definitions of helper functions use a C++11 features called
> +  // alias template. This feature can be mimiked by non C++11 support simply by
> +  // defining a type that inherits from the template we need to alias.  However,
> +  // the non-C++11 code requires the (re)definition of all constructors of the
> +  // derived class. The non-C++11 code follows the commented out C++11 code.
> +  // When we move to C++11 we can use the more elgant code.
> +#if 0
>
> There are some macros in CGAL to enable the code on platforms that support
> template aliases.
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arr_spherical_topology_traits_2.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125839421>:
>
> > -  typedef Sweep_line_construction_visitor
> -    Sweep_line_non_intersecting_construction_visitor;
> -
> -  typedef Arr_basic_insertion_sl_visitor<BIHelper>
> -    Sweep_line_non_intersecting_insertion_visitor;
> -
> -  template <typename OutputIterator_>
> -  struct Sweep_line_batched_point_location_visitor :
> -    public Arr_batched_pl_sl_visitor<BplHelper, OutputIterator_>
> +  // The following definitions of helper functions use a C++11 features called
> +  // alias template. This feature can be mimiked by non C++11 support simply by
> +  // defining a type that inherits from the template we need to alias.  However,
> +  // the non-C++11 code requires the (re)definition of all constructors of the
> +  // derived class. The non-C++11 code follows the commented out C++11 code.
> +  // When we move to C++11 we can use the more elgant code.
> +#if 0
>
> Same here
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arr_unb_planar_topolog
> y_traits_2.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125840549>:
>
> > -  typedef Sweep_line_construction_visitor
> -    Sweep_line_non_intersecting_construction_visitor;
> -
> -  typedef Arr_basic_insertion_sl_visitor<BIHelper>
> -    Sweep_line_non_intersecting_insertion_visitor;
> -
> -  template <class OutputIterator_>
> -  struct Sweep_line_batched_point_location_visitor :
> -    public Arr_batched_pl_sl_visitor<BplHelper, OutputIterator_>
> +  // The following definitions of helper functions use a C++11 features called
> +  // alias template. This feature can be mimiked by non C++11 support simply by
> +  // defining a type that inherits from the template we need to alias.  However,
> +  // the non-C++11 code requires the (re)definition of all constructors of the
> +  // derived class. The non-C++11 code follows the commented out C++11 code.
> +  // When we move to C++11 we can use the more elgant code.
> +#if 0
>
> same here
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arrangement_2/Arrangem
> ent_on_surface_2_global.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125840974>:
>
> >
> +#include <CGAL/license/Arrangement_on_surface_2.h>
>
> The license include directive should be right after the include guards
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Arrangement_2/Arrangem
> ent_on_surface_2_global.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125841241>:
>
> > @@ -182,7 +187,7 @@ void insert(Arrangement_on_surface_2<GeomTraits, TopTraits>& arr,
>      std::cout << "xxxxxxxxxxxx" << std::endl;
>    }
>  */
> -  arr_zone.init (c, pl);
> +  arr_zone.init(c, pl);
>
> Since you are cleaning the files, any reason to keep the part with the *Needs
> to be deleted!!!* flag?
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_
> construction_subcurve.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125843322>:
>
> > +  typedef GeometryTraits_2                              Geometry_traits_2;
> +  typedef Subcurve_                                     Subcurve;
> +  typedef Event_                                        Event;
> +  typedef Allocator_                                    Allocator;
> +
> +private:
> +  typedef Geometry_traits_2                             Gt2;
> +  typedef SurfaceSweepBaseCurve<Gt2, Event, Allocator, Subcurve>
> +                                                        Base;
> +
> +public:
> +  typedef typename Gt2::X_monotone_curve_2              X_monotone_curve_2;
> +  typedef Event*                                        Event_ptr;
> +  typedef std::list<unsigned int>                       Halfedge_indices_list;
> +
> +  /*! Construct deafult. */
>
> typo
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_
> insertion_sl_visitor.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125843448>:
>
> > +   */
> +  virtual bool is_split_event(Subcurve* sc, Event* event);
> +
> +  /*! Split the given edge edge.
> +   * \param he The edge to split.
> +   * \param sc The associated subcurve.
> +   * \param The split point.
> +   * \return A handle to the split edge.
> +   */
> +  virtual Halfedge_handle split_edge(Halfedge_handle he, Subcurve* sc,
> +                                     const Point_2& pt);
> +  //@}
> +};
> +
> +//-----------------------------------------------------------------------------
> +// Memeber-function definitions:
>
> typo
> ------------------------------
>
> In Arrangement_on_surface_2/include/CGAL/Surface_sweep_2/Arr_
> no_intersection_insertion_sl_visitor.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125843831>:
>
> >   * A sweep-line visitor for inserting new curves into an existing arrangement
>   * embedded on a surface, where these curves are interior-disjoint from all
>   * existing arrangement edges and vertices (so no intersections occur).
>   */
> -template <typename Helper_>
> -class Arr_basic_insertion_sl_visitor : public Helper_::Parent_visitor {
> +template <typename Helper_, typename Visitor_ = Default>
> +class Arr_no_intersection_insertion_sl_visitor :
>
> General remark: you might want to replace _sl_ by _ss_ everywhere.
> ------------------------------
>
> In Documentation/doc/Documentation/dependencies
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125844985>:
>
> > @@ -57,7 +57,7 @@ Principal_component_analysis
>  STL_Extension
>  Skin_surface_3
>  Snap_rounding_2
> -Sweep_line_2
> +Surface_sweep_2
>
> This change will require a redirect in the apache file of the doc so that
> latest will be correctly redirected.
> ------------------------------
>
> In Surface_sweep_2/examples/Surface_sweep_2/plane_sweep.cpp
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125845582>:
>
> >  #include <list>
>
> -typedef CGAL::Quotient<CGAL::MP_Float>                  NT;
> -typedef CGAL::Cartesian<NT>                             Kernel;
> -typedef Kernel::Point_2                                 Point_2;
> -typedef CGAL::Arr_segment_traits_2<Kernel>              Traits_2;
> -typedef Traits_2::Curve_2                               Segment_2;
> +#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
> +#include <CGAL/Arr_segment_traits_2.h>
> +#include <CGAL/Surface_sweep_2_algorithms.h>
>
> The header renaming is a breaking change that must be announced in
> Installation.html + the former header should still be kept with a redirect
> to the new header. You can use macros for this purpose such as in
> AABB_tree/include/CGAL/AABB_polyhedron_triangle_primitive.h for example.
> ------------------------------
>
> In Surface_sweep_2/include/CGAL/Surface_sweep_2/Curve_pair.h
> <https://github.com/CGAL/cgal/pull/2236#discussion_r125845845>:
>
> >    {
> -    // The smallest pointer will be the first.
> -    if(sc1 < sc2)
> -      m_pair = std::make_pair (sc1,sc2);
> -    else
> -      m_pair = std::make_pair (sc2,sc1);
> +    // The smallest pointer will be the first.
> +    if (sc1 < sc2) m_pair = std::make_pair (sc1,sc2);
> +    else m_pair = std::make_pair (sc2,sc1);
>
> there is a function CGAL::make_sorted_pair() in CGAL/utility.h
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#pullrequestreview-48264533>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuKOH9ZTGxF5vBolCDpnrt1cHGBxUks5sLKL3gaJpZM4OOKw6>
> .
>

sloriot commented:
For 1 you can use `#if !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) && !defined(BOOST_NO_TEMPLATE_ALIASES)` to handle both older and newer versions of boost. Note that you can also introduce `CGAL_CFG_NO_CPP0X_TEMPLATE_ALIASES` in `Installation/include/CGAL/config.h` if you want.

For 2, this is something @lrineau will do when integrating the branch. I've added a special label to the PR so that we don't forget.

efifogel commented:
Thanks.
I've introduced CGAL_CFG_NO_CPP0X_TEMPLATE_ALIASES in
Installation/include/CGAL/config.h

The branch is ready for integration.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 9 July 2017 at 17:54, Sebastien Loriot <notifications@github.com> wrote:

> For 1 you can use #if !defined(BOOST_NO_CXX11_TEMPLATE_ALIASES) &&
> !defined(BOOST_NO_TEMPLATE_ALIASES) to handle both older and newer
> versions of boost. Note that you can also introduce
> CGAL_CFG_NO_CPP0X_TEMPLATE_ALIASES in Installation/include/CGAL/config.h
> if you want.
>
> For 2, this is something @lrineau <https://github.com/lrineau> will do
> when integrating the branch. I've added a special label to the PR so that
> we don't forget.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-313924760>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuIB8PYpcqPKSrZGgeeqCG7ObRSGzks5sMOm8gaJpZM4OOKw6>
> .
>

lrineau commented:
I have killed the Travis job, because this PR:
  - is not for CGAL-4.11,
  - and will probably be modified multiple times before it is integration into CGAL.

efifogel commented:
I don't see a reason not to introduce it to 4.11.
It is ready and I'm not aware of any more changes needed.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 12 July 2017 at 12:13, Laurent Rineau <notifications@github.com> wrote:

> I have killed the Travis job, because this PR:
>
>    - is not for CGAL-4.11,
>    - and will probably be modified multiple times before it is
>    integration into CGAL.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-314703911>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuHdQO2uSYQXnW50jlAaKrNwdOptUks5sNI4egaJpZM4OOKw6>
> .
>

lrineau commented:
I see several reasons:
1. the code freeze for CGAL-4.11 was two weeks ago, even before you created this PR,
2. 252 files are changed, and it cannot be considered as a small low-risk change,
3. the PR has never been tested so far, and until today it was failing in Travis, and I plan to finish CGAL-4.11-beta1 tomorrow.

Do you have a good reason to force it into CGAL-4.11?

lrineau commented:
115 files are actually modified, if we skip renamings without change of the file. That is still a lot.

efifogel commented:
This is the reason I want to get it out.
Then, other changes and fixes are done on top of it.
Anyhow, as I've said the PR is ready as far as I can tell. If you have
other pressing PRs I understand.



   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 12 July 2017 at 13:09, Laurent Rineau <notifications@github.com> wrote:

> 115 files are actually modified, if we skip renamings without change of
> the file. That is still a lot.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-314717431>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuFph4rv2AYg0BQvuH8qW-bHUZ7yBks5sNJtggaJpZM4OOKw6>
> .
>

lrineau commented:
That is not the others PR that are pressing but the schedule. Such big a change should be done early in the release cycle, and certainly not after the code freeze. For CGAL-4.12, the timing is perfect.

efifogel commented:
ok, no problem.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 13 July 2017 at 15:49, Laurent Rineau <notifications@github.com> wrote:

> That is not the others PR that are pressing but the schedule. Such big a
> change should be done early in the release cycle, and certainly not after
> the code freeze. For CGAL-4.12, the timing is perfect.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-315067745>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuAAqDq4bAYKRO8GDcU5J9kBt37eMks5sNhJMgaJpZM4OOKw6>
> .
>

sloriot commented:
@efifogel There are some new warnings. See [test results](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-4.shtml). I  haven't checked for new errors.

efifogel commented:
Thanks for pointing it out.
All relevant warnings have been handled.
The errors are unrelated.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 2 August 2017 at 15:29, Sebastien Loriot <notifications@github.com>
wrote:

> @efifogel <https://github.com/efifogel> There are some new warnings. See test
> results
> <https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-4.shtml>.
> I haven't checked for new errors.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-319657732>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuGeJI4UQR_mOGsNiIkpMr99-Th81ks5sUGu0gaJpZM4OOKw6>
> .
>

sloriot commented:
@efifogel  You still have warnings in Kernel_23, Boolean_operations_2, Minkowski_sum_2, and Polygon_Demo.
See [test results](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-6.shtml).

efifogel commented:
Thanks for watching my back.
All fixed now.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 4 August 2017 at 10:48, Sebastien Loriot <notifications@github.com>
wrote:

> @efifogel <https://github.com/efifogel> You still have warnings in
> Kernel_23, Boolean_operations_2, Minkowski_sum_2, and Polygon_Demo.
> See test results
> <https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-6.shtml>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-320181990>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuJKbRlRr0FUga_XkiJGBK3rSSYqnks5sUszPgaJpZM4OOKw6>
> .
>

sloriot commented:
One warning remains: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-9/Surface_sweep_2/TestReport_lrineau_CentOS7-ICC-Release.gz
It is a strange one. Maybe an issue with the testsuite script?

efifogel commented:
I don't even know how to produce it...

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 8 August 2017 at 14:31, Sebastien Loriot <notifications@github.com>
wrote:

> One warning remains: https://cgal.geometryfactory.
> com/CGAL/testsuite/CGAL-4.12-Ic-9/Surface_sweep_2/
> TestReport_lrineau_CentOS7-ICC-Release.gz
> It is a strange one. Maybe an issue with the testsuite script?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-320928855>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuBCU5j-79gK_lLMkJHn9GQH3Jx5-ks5sWEcqgaJpZM4OOKw6>
> .
>

sloriot commented:
The warning seems to be reported by ICC [see](http://www.itkeyword.com/doc/4956007848491213829/problems-setting-environment-variables-for-intel-c-composer-gcc-not-found)/ I don't know why only `Surface_sweep_2, `Snap_rounding_2`, and `Arrangement_on_surface_2` have the warning. But it is probably not a coincidence that theses test directories have custom `cgal_test_base` scripts. @lrineau @maxGimeno Any idea?

lrineau commented:
You are right: that is because the `cgal_test` scripts in those directories calls a sub-level bash, and that platform has `BASH_ENV=/etc/profile.d/intel.sh` in its environment. Our file `/etc/profile.d/intel.sh` should have a sort of guard variable, to prevent its reloading when it has already been evaluated once.

lrineau commented:
This PR has been left untouched for a month. Is that on purpose?

efifogel commented:
It's my fault.
I'll get to it in a couple of days.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 18 September 2017 at 13:27, Laurent Rineau <notifications@github.com>
wrote:

> This PR has been left untouched for a month. Is that on purpose?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-330181108>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuLlOthxV1LUjFnZzIznC_SochieYks5sjkV_gaJpZM4OOKw6>
> .
>

lrineau commented:
Thanks Efi. That's fine. I just wanted to avoid that it was untouched/stalled because of me.

lrineau commented:
Issues with the new SPDX identifiers (cc @sloriot):
- https://travis-ci.org/CGAL/cgal/jobs/303048767#L2493

sloriot commented:
Should be fixed

lrineau commented:
https://travis-ci.org/CGAL/cgal/jobs/305259095#L2497
```
The following files do not have a SPDX license identifier:
Surface_sweep_2/include/CGAL/Surface_sweep_2/Subcurves_visitor.h
```

lrineau commented:
This still fails in Travis: https://travis-ci.org/CGAL/cgal/jobs/305772090#L6819

efifogel commented:
ok, I've renamed some left overs sweep_line with surface_sweep.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/



On 29 November 2017 at 16:28, Laurent Rineau <notifications@github.com>
wrote:

> This still fails in Travis: https://travis-ci.org/CGAL/
> cgal/jobs/305772090#L6819
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/2236#issuecomment-347876355>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGoWuGve-VF7MtMF2wcLwetnaKbmuv11ks5s7WoRgaJpZM4OOKw6>
> .
>

MaelRL commented on .travis.yml:
```diff
@@ -1,59 +1,59 @@
-language: cpp 
```
This file (`.travis.yml`) and `.travis/package.txt` are generated with scripts. If the trailing whitespace is removed, the script should be updated. Otherwise, if you rerun the script, the trailing whitespace is back and you get the inverse diff.

sloriot commented:
Tested in [CGAL-4.12-Ic-124](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-124)

@lrineau  For the next release, we need to add a redirect in the apache file of the doc so that latest will be correctly redirected from `Sweep_line_2` to `Surface_sweep_2`. I open issue #2699
